contracts/crowdsale/emission/PremintedCrowdsale.sol

```diff
@@ -0,0 +1,21 @@
+pragma solidity ^0.4.18;
+
+import "../Crowdsale.sol";
+import "../../token/ERC20/ERC20.sol";
+
+contract PremintedCrowdsale is Crowdsale {
+
+  address public tokenWallet;
+
+  function PremintedCrowdsale(address _tokenWallet) {
+    require(_tokenWallet != address(0));
+    tokenWallet = _tokenWallet;
+  }
+
+  // TODO: consider querying approval left and end crowdsale if depleted
+  // But approval could be increased..
```

Comment:
See https://github.com/OpenZeppelin/zeppelin-solidity/blob/master/contracts/token/ERC20/StandardToken.sol#L59

Reply:
I'm not sure I understand this, my point in that reply was that maybe the wallet is depleted, but then the allowance could be increased by [`increaseApproval()`](https://github.com/OpenZeppelin/zeppelin-solidity/blob/master/contracts/token/ERC20/StandardToken.sol#L73) and the crowdsale continue, right? Do we not want to allow this?

Reply:
The function I have in mind simply returns the current allowance.
```
function remainingTokens() public returns (uint256) {
  return token.allowance(_tokenWallet, this);
}
```

Reply:
Great, I'll put that in.